javaagent/build.gradle.kts

```diff
@@ -422,17 +425,30 @@ fun CopySpec.copyByteBuddy(jar: Provider<RegularFile>) {
   }
 }
 
+// exclude bootstrap projects from javaagent libs - they won't be added to inst/
+fun ShadowJar.excludeNonBootstrapClasses() {
+  // exclude the agent part of the javaagent-extension-api; these classes will be added in relocate tasks
+  exclude("io/opentelemetry/javaagent/extension/**")
+  exclude("**/instrumentation/api/incubator/sdk/**")
+}
+
 // exclude bootstrap projects from javaagent libs - they won't be added to inst/
 fun ShadowJar.excludeBootstrapClasses() {
   dependencies {
     exclude(project(":instrumentation-api"))
-    exclude(project(":instrumentation-api-incubator"))
     exclude(project(":instrumentation-annotations-support"))
     exclude(project(":javaagent-bootstrap"))
   }
 
   // exclude the bootstrap part of the javaagent-extension-api
   exclude("io/opentelemetry/javaagent/bootstrap/**")
+
+  // all in instrumentation-api-incubator except the bridge package
+  exclude("io/opentelemetry/instrumentation/api/incubator/builder/**")
+  exclude("io/opentelemetry/instrumentation/api/incubator/config/**")
+  exclude("io/opentelemetry/instrumentation/api/incubator/instrumenter/**")
+  exclude("io/opentelemetry/instrumentation/api/incubator/log/**")
+  exclude("io/opentelemetry/instrumentation/api/incubator/semconv/**")
```

Comment:
if a new package was added and we forgot to add an exclusion for it here, would something fail in a way we'd know to add it?

Reply:
I believe this would fail https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/7a69707f1f67cf797ae17a50c7f2f880bcd0f09b/testing-common/integration-tests/src/test/java/instrumentation/AgentInstrumentationTest.java#L61

It failed for me until I changed this at least.

Reply:
cool, i was wondering how that change was related